### PR TITLE
Add maxent test

### PR DESCRIPTION
**Pull request by Jiayi-Wang-Joey**

### CODE REVIEW

File: run_max_ent.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+"""Script running the MaxEnt IRL
+algorithm on a trees data set.
+Usage:
+    python scripts/run_max_ent.py data=$PWD/path_to_trees.csv n_action=5
+Note:
+    The data path must be absolute not relative.
+"""
+import dataclasses
+from locale import normalize
```
Comment (pawel-czyz) [inline run_max_ent.py:9]: Why do we need locales here?

File: run_max_ent.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+from typing import Any, Dict, Optional
+
+import anytree
+from matplotlib.pyplot import get
+import pandas as pd
+import numpy as np
+
+import phirl.api as ph
+import phirl.hydra_utils as hy
+from phirl.maxent import *
```
Comment (pawel-czyz) [inline run_max_ent.py:20]: The syntax `from module import *` is considered a bad practice in modern Python (namespace collisions and harder dependencies tracking essentially: https://stackoverflow.com/a/2360808).

File: run_max_ent.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+from irl_maxent import optimizer as O
```
Comment (pawel-czyz) [inline run_max_ent.py:21]: I'd suggest just `from irl_maxent import optimizer` – `O` is easy to confuse with `0`, plus this import syntax (importing as a single uppercase letter) is quite rare. (The only exception to this rule is `pytorch.nn.functional`, but I really don't know why it's that).

File: run_max_ent.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    logger.info("Setting up featurizier...") 
+    featurizer = IdentityFeaturizer(config.n_action)
+    features = get_features(featurizer, state_space)
+    logger.info(f"Feature dimensionality: {featurizer.shape}")
+
+    logger.info("Extracting trajectories...") 
+    action_of_trajectories = get_action_of_trajectories(trees, max_length=20)
+    n_states = get_n_states(config.n_action)
+    initial_state = tuple([0,0,0,0,0])
```
Comment (pawel-czyz) [inline run_max_ent.py:73]: This initial state should actually be defined via `config.n_actions`.

Something like `tuple(0 for _ in range(config.n_actions))` should work.

File: run_max_ent.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    state_trajectories = get_state_transition_trajectories(trajectories)
+    
+
+    logger.info("Calculating feature expectations...")
+    counts = expected_empirical_feature_counts(mdp=mdp, featurizer=featurizer, trajectories=state_trajectories)
+    feature_expectation = sum(np.array(list(counts.values())))
+    logger.info(f"Feature expectation: {feature_expectation}")
+    
+    logger.info("Learning the state reward function...")
+    optim = O.ExpSga(lr=O.linear_decay(lr0=0.2), normalize=True)
```
Comment (pawel-czyz) [inline run_max_ent.py:84]: I think that the learning rate should be a parameter in the Hydra config. (Similarly as `n_actions` is).

File: run_max_ent.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    p_initial = np.zeros(n_states)
+    p_initial[0] = 1
+    
+    
+    s_reward, _ = irl(n_actions=config.n_action, features=features, feature_expectation=feature_expectation, optim = optim,
```
Comment (pawel-czyz) [inline run_max_ent.py:89]: Similarly as mentioned above, `eps` and `eps_esvf` should be in Hydra config.

File: tests/phirl/test_maxent.py [hunk 7]
```diff
@@ -14,7 +22,7 @@ def test_n_states(n_actions: int) -> None:
 
 
 @pytest.mark.parametrize("n_actions", (2, 5, 7))
-def test_state_space(n_actions: int) -> None:
+def test_state_space(n_actions) -> None:
```
Comment (pawel-czyz) [inline tests/phirl/test_maxent.py:25]: It's a nit, but I'd keep the `:int` here.

**Review by pawel-czyz — CHANGES_REQUESTED** [review]
Hi Jiayi,

I have had several comments (mostly minor), but overall I am very happy with your code! My officemate was asking if you'd be interested in switching to robotics, as they need people who can write such a good code :wink: 
Wonderful job!

File: run_max_ent.py (hunk 3 above)
Comment (pawel-czyz) [inline run_max_ent.py:21]: Actually, do we have `irl_maxent` listed as a dependency? (If not, it'll require adding the `irl_maxent` line in `requirements/basic.txt` file).

**Comment by Jiayi-Wang-Joey** [comment]
Pawel, thank you for reviewing the code and I have improved the code. (And thank your officemate! I am really interested in robotics haha, however, they are too competitive:laughing:)

File: run_max_ent.py (hunk 3 above)
Comment (Jiayi-Wang-Joey) [reply run_max_ent.py]: I think we have `irl-maxent` but not `irl_maxent`.
